haigha/reader.py

```diff
@@ -232,7 +232,7 @@ def read_timestamp(self):
         Will raise BufferUnderflow if there's not enough bytes in the buffer.
         Will raise struct.error if the data is malformed
         """
-        return datetime.utcfromtimestamp(self.read_longlong())
+        return datetime.utcfromtimestamp(self.read_longlong() / 1e6)
```

Comment:
This change doesn't seem right; in AMQP 0-9-1, timestamp is defined as follows per https://www.rabbitmq.com/resources/specs/amqp0-9-1.pdf:

> Time stamps are held in the 64-bit POSIX time_t format with an accuracy of one second. By using 64 bits
we avoid future wraparound issues associated with 31-bit and 32-bit time_t values.

IIRC, this becomes milliseconds in 0-10-0, but is seconds in 0-9-1. This PR divides the value by 1000000, which makes it appear as if the timestamp is neither seconds, nor milliseconds, but in microseconds.